zyte_common_items/pipelines.py

```diff
@@ -53,3 +57,49 @@ def __init__(self):
 
     def process_item(self, item, spider):
         return ae.downgrade(item)
+
+
+class DropItem(Exception):
+    """Drop item from the item pipeline."""
+
+    pass
+
+
+class DropLowProbabilityItemPipeline:
+    """Drop item with a probability less than threshold."""
+
+    def __init__(self, crawler):
+        self.stats = crawler.stats
+        self.thresholds = {}
+        self.init_thresholds(crawler.spider)
+
+    @classmethod
+    def from_crawler(cls, crawler):
+        o = cls(crawler)
+        return o
+
+    def init_thresholds(self, spider):
+        thresholds_settings = spider.settings.get("ITEM_PROBABILITY_THRESHOLDS", {})
+        for item, threshold in thresholds_settings.items():
+            item_type = item if isinstance(item, str) else global_object_name(item)
```

Comment:
I’m thinking my suggestion to use `global_object_name` was misguided.

Using [`load_object`](https://github.com/scrapy/scrapy/blob/1d11ea3a54607b436f9a88f07911902a4882f0e8/scrapy/utils/misc.py#L53) if `str` may be better, as it later allows to implement `get_threshold` without `global_object_name` for better efficiency and readability.

Reply:
Agree, nice function, trying to found something like that but couldn't. Thanks!